### PR TITLE
Retrieve version of package despite cloning method

### PR DESCRIPTION
**Pull request by nmuesch**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md).

### What does this PR do?

This PR adds in the ability to retrieve the version of this package, even when using a shallow clone. https://github.com/pypa/setuptools_scm/issues/ #93 The issue here is that setuptools_scm uses git tags to retrieve the version of the package. However, if you install this from a shallow clone, such as in a CI environment, you're left with a version of `0.1`

### Alternate Designs

An alternate approach could be to manually set this version during the release cycle in the setup.py file. However, this comes with the drawback that if you are using a development version to generate clients, you'll wind up with a non reproducible .apigentools-info file. 

### Verification Process

Performed a fresh install from a shallow clone and the version showed up as expected. 

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,10 +18,16 @@
 __version__ = "{version}"
 """
 
+
+def parse_fetch_on_shallow(root):
+    return parse(root, pre_parse=fetch_on_shallow)
```
Comment (jirikuncar) [inline setup.py:23]:
```suggestion
    from setuptools_scm.git import parse, fetch_on_shallow
    return parse(root, pre_parse=fetch_on_shallow)
```

File: setup.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 import os
 
 from setuptools import setup
+from setuptools_scm.git import parse, fetch_on_shallow
+
```
Comment (jirikuncar) [inline setup.py:10]:
```suggestion
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires = ["setuptools", "setuptools_scm"]
```
Comment (jirikuncar) [inline pyproject.toml:2]: do we have it already in `setup.cfg`? maybe we should remove it from there

File: setup.py (hunk 1 above)
Comment (jirikuncar) [reply setup.py:23]: avoid global imports in setup.py

File: pyproject.toml (hunk 3 above)
Comment (nmuesch) [reply pyproject.toml:2]: Yeah, I'm not sure the exact issue, but the way its installed through things like `pre-commit` isn't actually installing the setup requirements and it errors out. Will remove that section then and move it into pyproject.toml 👍

File: setup.py [hunk 4]
```diff
@@ -16,10 +16,17 @@
 __version__ = "{version}"
 """
 
+
+def parse_fetch_on_shallow(root):
+    from setuptools_scm.git import parse, fetch_on_shallow
+    return parse(root, pre_parse=fetch_on_shallow)
```
Comment (jirikuncar) [inline setup.py:22]:
```suggestion

    return parse(root, pre_parse=fetch_on_shallow)
```

**Pull request merged**